com.ibm.streamsx.topology/opt/python/packages/streamsx/topology/topology.py

```diff
@@ -232,15 +235,45 @@ def parallel(self, width):
         
         Args:
             width (int): degree of parallelism
+            routing - denotes what type of tuple routing to use. 
+                ROUND_ROBIN: delivers tuples in round robin fashion to downstream operators
+                HASH_PARTIONED: delivers to downstream operators based on the hash of the tuples being sent
+                KEY_PARTIONED: delivers to downstream operators based on a hash provided by the user function provided
+            func - User defined function called when KEY_PARTIONED routing is specified.  The function provides an
```

Comment:
You are right....I guess I am not clear on what the design point is for KEY_PARTITIONED.  How about I remove KEY_PARTITIONED for now and just include HASH_PARTITIONED with an optional user provided hash.  Then we can figure out what we really want for KEY_PARTITIONED